File: android/tutorials/tutorial2-composing-ribs-starter-code/src/test/java/com/uber/rib/root/logged_out/LoggedOutRouterTest.java
package com.uber.rib.root.logged_out;

import com.uber.rib.core.RibTestBasePlaceholder;
import com.uber.rib.core.RouterHelper;

import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

public class LoggedOutRouterTest extends RibTestBasePlaceholder {

    @Mock LoggedOutBuilder.Component component;
    @Mock LoggedOutInteractor interactor;
    @Mock LoggedOutView view;

    private LoggedOutRouter router;

    @Before
    public void setup() {
        MockitoAnnotations.initMocks(this);

        router = new LoggedOutRouter(view, interactor, component);
    }

    /**
     * TODO: Delete this example and add real tests.
     */
    @Test
    public void anExampleTest_withSomeConditions_shouldPass() {
        // Use RouterHelper to drive your router's lifecycle.
        RouterHelper.attach(router);
        RouterHelper.detach(router);

        throw new RuntimeException("Remove this test and add real tests.");
    }

}

File: android/demos/compose/src/test/kotlin/com/uber/rib/compose/root/main/logged_out/LoggedOutRouterTest.kt
package com.uber.rib.compose.root.main.logged_out

import com.uber.rib.core.RibTestBasePlaceholder
import com.uber.rib.core.RouterHelper

import org.junit.Before
import org.junit.Test
import org.mockito.Mock
import org.mockito.MockitoAnnotations

class LoggedOutRouterTest : RibTestBasePlaceholder() {

  @Mock internal lateinit var component: LoggedOutBuilder.Component
  @Mock internal lateinit var interactor: LoggedOutInteractor
  @Mock internal lateinit var view: LoggedOutView

  private var router: LoggedOutRouter? = null

  @Before
  fun setup() {
    MockitoAnnotations.initMocks(this)

    router = LoggedOutRouter(view, interactor, component)
  }

  /**
   * TODO: Delete this example and add real tests.
   */
  @Test
  fun anExampleTest_withSomeConditions_shouldPass() {
    // Use RouterHelper to drive your router's lifecycle.
    RouterHelper.attach(router!!)
    RouterHelper.detach(router!!)

    throw RuntimeException("Remove this test and add real tests.")
  }

}

